### DISCUSSION TITLE
How to make id "safe" on the typings?

### DISCUSSION THREAD

**Discussion by henriqueleite42** [prisma/prisma#22108] [Q&A]
### Question

Hi, I have the following schema:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Account {
  id        String   @id @db.Char(16)
  email     String?  @unique @db.VarChar(150)
  createdAt DateTime @default(now())
 }
```

And I want to **MANUALLY** generate the ID instead of leaving it for Prisma or the database, but when I do something like this:

```ts
prisma.account.create({
  data: {
    id: "my-custom-id",
    email: "foo@bar.com",
  },
})
```

I get this error:
```
Types of property 'id' are incompatible.
        Type 'string' is not assignable to type 'never'.ts(2322)
```

Is there a way to fix this, and make the typings understand that the `id` is a **required** field, that **always** has to be present?

### Expected behavior (optional)

Work as described here:
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#single-field-ids-without-default-values

**Comment by ludralph** [comment]
Hi @henriqueleite42 👋 

Thank you for raising this question.

I'm unable to reproduce the error you specified as I can correctly create the `id` using the custom `id` you passed. The error message you're seeing suggests that there is a type mismatch between the expected type for the `id` field and the type you're providing (`string`). This could be due to the Prisma schema not being properly reflected in the generated Prisma Client, did you generate the client by using the command `npx prisma generate` after a schema change?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
